sql-plugin/src/main/scala/org/apache/spark/sql/rapids/execution/python/GpuFlatMapGroupsInPandasExec.scala

```diff
@@ -98,34 +97,77 @@ case class GpuFlatMapGroupsInPandasExec(
   override def requiredChildOrdering: Seq[Seq[SortOrder]] =
     Seq(groupingAttributes.map(ShimLoader.getSparkShims.sortOrder(_, Ascending)))
 
-  override protected def doExecute(): RDD[InternalRow] = {
-    lazy val isPythonOnGpuEnabled = GpuPythonHelper.isPythonOnGpuEnabled(conf)
-    val inputRDD = child.execute()
+  private val pandasFunction = func.asInstanceOf[GpuPythonUDF].func
+
+  // One batch as input to keep the integrity for each group
+  override def childrenCoalesceGoal: Seq[CoalesceGoal] = Seq(RequireSingleBatch)
```

Comment:
Can we have a follow on issue to fix this? We are very much trying to move away from anything that requires a single batch. For windowing we are going to have to come up with a good iterator that can split batches on key boundaries, so it is not something that we have to do right now. But we should look at supporting it eventually.

Reply:
Sure, here it is https://github.com/NVIDIA/spark-rapids/issues/2200 .

It will be better to let this 'good' iterator support setting the number of the unit (e.g. group) in each split batch. Then it can replace the `BatchGroupedIterator` by setting the number to `1`.